src/YesSql.Provider.Common/SqlBuilder.cs

```diff
@@ -198,6 +207,17 @@ public virtual string ToSqlString()
 
                 sb.Append("SELECT ");
 
+                if (_distinct != null)
+                {
+                    sb.Append(_distinct);
+
+                    if (_order != null && _order[0] != "[Id]")
```

Comment:
If this is to fix another unit test, I want to understand which one it is, because that seems super hacky

Reply:
It is hacky... and yes it's to fix another test.  I'm still working on it....

The problem is some queries do an Order By on Columns that aren't in the Select.  
But when DISTINCT used, then all Order By columns must be in the Select

Reply:
But I can't just add all the Order Clauses to the Select, because some break it in other ways

Reply:
What database requires the select to have the order by field in the select clause?

Reply:
SQLServer does.  SQLite didn't seem to care.  I haven't been able to test other dbs locally